### ISSUE TITLE
Unknown->Any through flow execution (like match-case) same as isinstance method

### ISSUE THREAD

**Issue by valentino-sm** [microsoft/pyright#5939]
**Describe the bug**

I guess that Unknown->Any through flow execution should be resolved same as isinstance method.
```python
def reportUnknownArgumentType(value: Sequence[Any] | Mapping[Any, Any] | Any) -> None:
    if isinstance(value, Sequence):
        reveal_type(value)  # Sequence[Any]
    if isinstance(value, dict):
        reveal_type(value)  # dict[Any, Any]
    if isinstance(value, list):
        reveal_type(value)  # list[Any]
    match value:
        case dict():
            reveal_type(value)  # dict[Any, Any] | dict[Unknown, Unknown]
        case list():
            reveal_type(value)  # list[Any] | list[Unknown]
        case _:
            reveal_type(value)  # Sequence[Any] | Mapping[Any, Any] | Any
    reveal_type(
        value
    )  # dict[Any, Any] | dict[Unknown, Unknown] | list[Any] | list[Unknown] | Sequence[Any] | Mapping[Any, Any] | Any
    if isinstance(value, dict):
        reveal_type(value)  # dict[Any, Any] | dict[Unknown, Unknown]
```
pyright 1.1.327

**Comment by erictraut** [comment]
I agree there is an inconsistency between the type narrowing logic for class patterns and `isinstance`. The class pattern type narrowing logic is technically more correct, but changing the `isinstance` logic to match it would create a lot of noise in code bases that use pyright in strict mode. I'm therefore inclined to leave the current behavior as is for now.

If you're not using strict mode, the inconsistency doesn't matter because `Any` and `Unknown` are treated the same. If you are using strict mode, then you probably care more about type soundness in your code, and I'd encourage you to eliminate the use of `Any` where possible, especially a "naked" `Any` in the type for `value`.

**Comment by erictraut** [comment]
I'm going to leave the behavior as is for now. There's a good chance we'll revisit this in the future as part of a consistency pass for type narrowing — perhaps even across type checkers if we can get agreement on the preferred behavior. If and when we do this, it will cause some disruption. I'd prefer to do that once rather than in a piecemeal manner. I'm going to close this for now.
